tests/bsim/bluetooth/audio/src/public_broadcast_source_test.c

```diff
@@ -0,0 +1,381 @@
+/*
+ * Copyright 2023 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#if defined(CONFIG_BT_PBP)
+
+#include <zephyr/bluetooth/bluetooth.h>
+#include <zephyr/bluetooth/audio/audio.h>
+#include <zephyr/bluetooth/audio/bap_lc3_preset.h>
+#include <zephyr/bluetooth/audio/bap.h>
+#include <zephyr/bluetooth/audio/cap.h>
+#include <zephyr/bluetooth/audio/pbp.h>
+
+#include "common.h"
+
+/* When BROADCAST_ENQUEUE_COUNT > 1 we can enqueue enough buffers to ensure that
+ * the controller is never idle
+ */
+#define BROADCAST_ENQUEUE_COUNT 2U
+#define TOTAL_BUF_NEEDED	(BROADCAST_ENQUEUE_COUNT * CONFIG_BT_BAP_BROADCAST_SRC_STREAM_COUNT)
+/* PBS ASCII text */
+#define PBS_DEMO                0x50, 0x42, 0x50
+
+extern enum bst_result_t bst_result;
+
+BUILD_ASSERT(CONFIG_BT_ISO_TX_BUF_COUNT >= TOTAL_BUF_NEEDED,
+	     "CONFIG_BT_ISO_TX_BUF_COUNT should be at least "
+	     "BROADCAST_ENQUEUE_COUNT * CONFIG_BT_BAP_BROADCAST_SRC_STREAM_COUNT");
+
+NET_BUF_POOL_FIXED_DEFINE(tx_pool,
+			  TOTAL_BUF_NEEDED,
+			  BT_ISO_SDU_BUF_SIZE(CONFIG_BT_ISO_TX_MTU),
+			  CONFIG_BT_CONN_TX_USER_DATA_SIZE, NULL);
+
+const uint8_t pba_metadata[] = {BT_AUDIO_CODEC_DATA(
+	BT_AUDIO_METADATA_TYPE_PROGRAM_INFO, PBS_DEMO)};
+
+static uint8_t bis_codec_data[] = {BT_AUDIO_CODEC_DATA(
+	BT_AUDIO_CODEC_CONFIG_LC3_FREQ, BT_BYTES_LIST_LE16(BT_AUDIO_CODEC_CONFIG_LC3_FREQ_48KHZ))};
```

Comment:
```suggestion
static uint8_t bis_codec_data[] = {
	BT_AUDIO_CODEC_DATA(BT_AUDIO_CODEC_CONFIG_LC3_FREQ,
			    BT_BYTES_LIST_LE16(BT_AUDIO_CODEC_CONFIG_LC3_FREQ_48KHZ))
};
```

Does PBP mandate that we put the frequency in the BIS CC data? Isn't it enough to have it at the subgroup level?